src/library.cpp

```diff
@@ -296,14 +296,18 @@ Library::BookIdCollection Library::getBooksByTitleOrDescription(const Filter& fi
   queryParser.set_default_op(Xapian::Query::OP_AND);
   queryParser.add_prefix("title", "S");
   queryParser.add_prefix("description", "XD");
+  const auto partialQueryFlag = filter.queryIsPartial()
+                              ? Xapian::QueryParser::FLAG_PARTIAL
+                              : 0;
   // Language assumed for the query is not known for sure so stemming
   // is not applied
   //queryParser.set_stemmer(Xapian::Stem(iso639_3ToXapian.at(???)));
   //queryParser.set_stemming_strategy(Xapian::QueryParser::STEM_SOME);
   const auto flags = Xapian::QueryParser::FLAG_PHRASE
                    | Xapian::QueryParser::FLAG_BOOLEAN
                    | Xapian::QueryParser::FLAG_LOVEHATE
-                   | Xapian::QueryParser::FLAG_WILDCARD;
+                   | Xapian::QueryParser::FLAG_WILDCARD
+                   | partialQueryFlag;
```

Comment:
I would have directly set `flags` with the default (always present) flags and them update it if needed :
```cpp
if (filter.queryIsPartial()) {
  flags |= Xapian::QueryParser::FLAG_PARTIAL;
}

Reply:
I prefer to stick to the immutable state paradigm whenever possible

Reply:
It is a acceptable argument.